### PR TITLE
Integrate CISA review feedback into aad baseline

### LINKED ISSUE #532: Integrate CISA review feedback into AAD baseline
# 💡 Summary #

CISA formal review will provide feedback that may require changes to the markdown formatted Azure AD baselines.

## Motivation and context ##

CISA formal review may discover changes that materially affect the baselines.  If changes are made, they should be reflected in the markdown baselines as the source of truth for secure configuration baselines.

## Implementation notes ##

Please provide details for implementation, such as:

- Review CISA feedback in terms of suggested changes and comments
- Adjudicate comments with CISA to confirm recommended changes
- Make recommended changes to the markdown baselines

## Acceptance criteria ##

How do we know when this work is done?

- [x] CISA feedback on Azure AD SCB has been received by the developers
- [x] All CISA feedback has been reviewed and changes recommended and adjudicated successfully
- [ ] Changes have been integrated into Azure AD markdown and merged into Emerald branch

### PR DESCRIPTION
**Pull request by tkol2022**
#  #

## 🗣 Description ##

This update includes the following changes to the M365 AAD security configuration baseline:

- Changes to improve overall consistency of format and language throughout
- Grammar and spelling fixes
- Language and verbiage updates for increased reader clarity

This update does NOT include any materially significant changes to the number of or content in policy items in the baseline itself that would require a change in settings to align with the baseline recommendations.

## 💭 Motivation and context ##

The updates included further refine and improve the overall readability and clarity of the baseline document.  The result should be a baseline that is easier to interpret and use as a reference for M365 tenant administrators.

Closes #532 

## 🧪 Testing ##

Compare the separately provided Word version with track changes of the AAD baseline document with the changes in the markdown in this branch.  Accepted/modified suggested changes as appropriate and mirrored in markdown, rejected and additional changes commented in Word and sync'd to markdown as needed.

Manual review of markdown and ensure that the automated markdown test on the PR passes.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [ ] Rebase the branch against the parent branch prior to merge as needed.
- [ ] All CISA feedback has been reviewed and adjudicated successfully

## ✅ Post-merge checklist ##

- [ ] Confirm that the parent branch passes all tests after the merge

### CODE REVIEW

File: baselines/aad.md [hunk 1]
```diff
@@ -649,40 +629,40 @@ Performing a manual review of highly privileged users to determine which ones ar
     Get-MgBetaUser -Filter "userPrincipalName eq 'username@somedomain.com'" | FL
```
Comment (nanda-katikaneni) [inline baselines/aad.md:629]: This displayed field is inconsistent with word document (which has an additional middle line with Select-Mgprofile beta)

File: baselines/aad.md [hunk 2]
```diff
@@ -628,19 +609,18 @@ The following implementation instructions that reference the AAD PIM service wil
...
 1. Perform the steps below for each highly privileged role. We reference the Global Administrator role as an example.
 
 2. In **Azure Active Directory** select **Roles and administrators.**
 
 3. Select the **Global administrator role.**
 
-4. Under **Manage**, select **Assignments.** If you have AAD PIM, repeat the steps below for both the **Eligible** and the **Active** assignments. If a group is listed, you will need to determine the members of the group and perform the steps for each group member.
+4. Under **Manage**, select **Assignments.** If you have Azure AD PIM, repeat the steps below for both the **Eligible** and the **Active** assignments. If a group is listed, you will need to determine the members of the group and perform the steps for each group member.
 
 5. For each highly privileged user, execute the Powershell code below but replace the `username@somedomain.com` with the principal name of the user who is specific to your environment. You can get the data value from the **Principal name** field displayed in the Azure Active Directory portal.
```
Comment (nanda-katikaneni) [inline baselines/aad.md:625]: MS.AAD.7.3v1 Instructions; step 5 - there is a reference to 'Azure Active Directory portal' which is inconsistent with word doc where its referred to 'Azure AD portal'

File: baselines/aad.md [hunk 3]
```diff
@@ -649,40 +629,40 @@ Performing a manual review of highly privileged users to determine which ones ar
...
+#### MS.AAD.7.4v1 Instructions
 
 1. In **Azure Active Directory** select **Roles and administrators**. Perform the steps below for each highly privileged role. We reference the Global Administrator role as an example.
 
 2. Select the **Global administrator role.**
 
 3. Under **Manage**, select **Assignments** and click the **Active assignments** tab.
 
-4. Verify there are no users or groups with a value of **Permanent** in the **End time** column. If there are any, recreate those assignments to have an expiration date using AAD PIM or an alternative PAM system. The only exception to this policy is emergency access accounts and service accounts requiring perpetual active assignments. See policy MS.AAD.7.4v1 note section for details.
+4. Verify there are no users or groups with a value of **Permanent** in the **End time** column. If there are any, recreate those assignments to have an expiration date using Azure AD PIM or an alternative PAM system. The only exception to this policy is emergency access accounts and service accounts requiring perpetual active assignments. See policy MS.AAD.7.4v1 note section for details.
```
Comment (nanda-katikaneni) [inline baselines/aad.md:642]: MS.AAD.7.4v1 Instructions - step 4 - the last sentence where "MS.AAD.7.4v1 note section" is referred - it does not have a hyperlink to point the user to note section (word doc has this).

**Review by nanda-katikaneni — CHANGES_REQUESTED** [review]
Looks good to most part - commented in-line about a few inconsistencies/mismatches. One comment about Word document: the table of contents need an update, it currently has a weird format. 

Will read through the full doc one more time for any additional comments.

File: baselines/aad.md [hunk 4]
```diff
@@ -1,34 +1,40 @@
...
 
 ## Key Terminology
 The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119).
 
 The following are key terms and descriptions used in this document.
 
-**Hybrid Azure Active Directory (AD)** – This term denotes the scenario
+**Hybrid Azure Active Directory (AD)**: This term denotes the scenario
 when an organization has an on-premises AD domain that contains the
 master user directory but federates access to the cloud Microsoft 365
```
Comment (ssatyapal123) [inline baselines/aad.md:30]: "Microsoft 365" is already expanded earlier. (The Word document removed this expansion.)

File: baselines/aad.md [hunk 5]
```diff
@@ -41,12 +47,9 @@ master user directory but federates access to the cloud Microsoft 365
```
Comment (ssatyapal123) [inline baselines/aad.md:47]:
```suggestion
Throughout this document, this list of highly privileged roles is referenced in numerous baseline policies. Agencies should consider this list a foundational reference and apply respective baseline policies to additional Azure AD roles as necessary.
```

File: baselines/aad.md [hunk 6]
```diff
@@ -41,12 +47,9 @@ master user directory but federates access to the cloud Microsoft 365
...
+As described in Microsoft’s literature related to conditional access policies, CISA recommends initially setting a policy to **Report-only** when it is created and then performing thorough hands-on testing to ensure that there are no unintended consequences before toggling the policy from **Report-only** to **On**. The policy will only be enforced when it is set to **On**. One tool that can assist with running test simulations is the [What If tool](https://learn.microsoft.com/en-us/azure/active-directory/conditional-access/what-if-tool). Microsoft also describes [Conditional Access insights and reporting features](https://learn.microsoft.com/en-us/azure/active-directory/conditional-access/howto-conditional-access-insights-reporting) that can assist with testing.
```
Comment (ssatyapal123) [inline baselines/aad.md:53]: In the Word document the text describing the link to "Conditional Access insights and reporting features" omits the word "features".

File: baselines/aad.md [hunk 7]
```diff
@@ -162,57 +164,58 @@ Sign-ins detected as high risk SHALL be blocked.
...
-Figure 1: Depiction of MFA methods from weakest to strongest. _Adapted from [MS Build Page](https://learn.microsoft.com/en-us/azure/active-directory/authentication/concept-authentication-methods)_
+Figure 1: Depiction of MFA methods from weakest to strongest. _Adapted from [Microsoft Page](https://learn.microsoft.com/en-us/entra/identity/authentication/concept-authentication-methods)_
 
 ### Policies
 #### MS.AAD.3.1v1
 Phishing-resistant MFA SHALL be enforced for all users.
 
 **Preferred phishing-resistant methods**
 
-The methods **AAD Certificate-Based Authentication (CBA)**, **FIDO2 Security Key** and **Windows Hello for Business** are the recommended options since they offer forms of MFA with the least weaknesses. AAD CBA supports Federal PIV cards when they authenticate directly to Azure AD.
```
Comment (ssatyapal123) [inline baselines/aad.md:178]: First letter of Federal is capitalized here but not in the Word document or in other parts of this markdown.

File: baselines/aad.md [hunk 8]
```diff
@@ -1,34 +1,40 @@
+**`TLP:CLEAR`**
+
 # CISA M365 Security Configuration Baseline for Azure Active Directory
 
-Azure Active Directory (AAD) is a cloud-based identity and access control service that provides security and functional capabilities to Microsoft 365. This Secure Configuration Baseline (SCB) provides specific policies to help secure AAD.
+Azure Active Directory (Azure AD) is a cloud-based identity and access control service that provides security and functional capabilities to Microsoft 365. This Secure Configuration Baseline (SCB) provides specific policies to help secure Azure AD.
```
Comment (mheller93766) [inline baselines/aad.md:5]: Microsoft 365 (M365) Azure Active Directory (Azure AD) is a cloud-based identity and access control service that provides security and functional capabilities. This Secure Configuration Baseline (SCB) provides specific policies to help secure Azure AD. 

Standardize front-matter across SCBs

**Review by mheller93766 — CHANGES_REQUESTED** [review]

File: baselines/aad.md [hunk 9]
```diff
@@ -228,39 +231,39 @@ Phishing-resistant MFA SHALL be required for highly privileged roles.
 Managed devices SHOULD be required for authentication.
 
 <!--Policy: MS.AAD.3.7v1; Criticality: SHOULD -->
-- _Rationale:_ The security risk of an adversary authenticating to the tenant from their own device is reduced by requiring a managed device to authenticate. Managed devices are under the provisioning and control of the agency. [OMB-22-09] (https://www.whitehouse.gov/wp-content/uploads/2022/01/M-22-09.pdf) specifically states, "When authorizing users to access resources, agencies must consider at least one device-level signal alongside identity information about the authenticated user."
+- _Rationale:_ The security risk of an adversary authenticating to the tenant from their own device is reduced by requiring a managed device to authenticate. Managed devices are under the provisioning and control of the agency. [OMB-22-09] (https://www.whitehouse.gov/wp-content/uploads/2022/01/M-22-09.pdf) states, "When authorizing users to access resources, agencies must consider at least one device-level signal alongside identity information about the authenticated user."
```
[the remark this answers is not in the document]
Comment (tkol2022) [reply baselines/aad.md:234]: Fixed.

File: baselines/aad.md (hunk 1 above)
Comment (tkol2022) [reply baselines/aad.md:629]: Fixed in the Word doc which was wrong. They match now.

File: baselines/aad.md (hunk 2 above)
Comment (tkol2022) [reply baselines/aad.md:625]: Fixed.

File: baselines/aad.md (hunk 3 above)
Comment (tkol2022) [reply baselines/aad.md:642]: This is one of the limitations of markdown. I cannot link to a section that doesn't have a # tag prefix. We might figure out a workaround in a future revision.

File: baselines/aad.md (hunk 4 above)
Comment (tkol2022) [reply baselines/aad.md:30]: Fixed.

File: baselines/aad.md (hunk 6 above)
Comment (tkol2022) [reply baselines/aad.md:53]: Fixed.

File: baselines/aad.md (hunk 7 above)
Comment (tkol2022) [reply baselines/aad.md:178]: This was fixed.

File: baselines/aad.md (hunk 8 above)
Comment (tkol2022) [reply baselines/aad.md:5]: Fixed.

File: baselines/aad.md [hunk 10]
```diff
@@ -27,14 +27,14 @@ The following are key terms and descriptions used in this document.
 
 **Hybrid Azure Active Directory (AD)**: This term denotes the scenario
 when an organization has an on-premises AD domain that contains the
-master user directory but federates access to the cloud Microsoft 365
+master user directory but federates access to the cloud M365
 (M365) Azure AD tenant.
```
Comment (mheller93766) [inline baselines/aad.md:31]: Remove (M365). Line 30 ends with M365. This is a repeat.

**Review by nanda-katikaneni — APPROVED** [review]

**Review by mheller93766 — APPROVED** [review]

**Review by ssatyapal123 — APPROVED** [review]

**Pull request merged**